adapters/goldbach/goldbach.go

```diff
@@ -0,0 +1,237 @@
+package goldbach
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type requestExtAdapter struct {
+	Goldbach requestExtGoldbach `json:"goldbach"`
+
+	*openrtb_ext.ExtRequest `json:"inline,omitempty"`
+}
+
+type requestExtGoldbach struct {
+	PublisherID  string `json:"publisherId"`
+	MockResponse *bool  `json:"mockResponse,omitempty"`
+}
+
+type impExtAdapter struct {
+	Goldbach impExtGoldbachOutgoing `json:"goldbach"`
+}
+
+type impExtGoldbachOutgoing struct {
+	Targetings map[string][]string `json:"targetings,omitempty"`
+	SlotID     string              `json:"slotId"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	if config.Endpoint == "" {
+		return nil, errors.New("missing endpoint adapter parameter")
+	}
+
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var reqs []*adapters.RequestData
+	var errs []error
+
+	var requestExt requestExtAdapter
+	if request.Ext != nil {
+		if err := jsonutil.Unmarshal(request.Ext, &requestExt); err != nil {
+			errs = append(errs, &errortypes.FailedToUnmarshal{Message: fmt.Errorf("unable to unmarshal request.ext: %w", err).Error()})
+		}
+	}
+
+	// group impressions by publisher ID
+	publisherImps := make(map[string][]openrtb2.Imp)
+	for _, imp := range request.Imp {
+		publisherID, impCopy, err := buildImp(imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		publisherImps[publisherID] = append(publisherImps[publisherID], impCopy)
+	}
+
+	if len(publisherImps) == 0 {
+		errs = append(errs, &errortypes.BadInput{Message: "no valid impression found"})
+	}
+
+	// create a separate request for each publisher
+	for publisherID, imps := range publisherImps {
+		requestPublisher, err := buildRequest(*request, publisherID, &imps, &requestExt)
```

Comment:
```suggestion
		requestPublisher, err := buildRequest(*request, publisherID, imps, &requestExt)
```
This only takes a pointer to the slice, it doesn't copy slice items